stores/metadata.go

```diff
@@ -92,24 +92,26 @@ type (
 		DBObjectID uint `gorm:"index"`
 
 		// Slice related fields.
-		Slab   dbSlab `gorm:"constraint:OnDelete:CASCADE"` // CASCADE to delete slabs too
-		Offset uint32
-		Length uint32
+		DBSlabID uint `gorm:"index"`
+		Offset   uint32
+		Length   uint32
 	}
 
 	dbSlab struct {
 		Model
-		DBSliceID uint `gorm:"index"`
 
 		Key         []byte `gorm:"unique;NOT NULL;size:68"` // json string
 		MinShards   uint8
 		TotalShards uint8
-		Shards      []dbShard `gorm:"constraint:OnDelete:CASCADE"` // CASCADE to delete shards too
+
+		Slices []dbSlice  `gorm:"constraint:OnDelete:SET NULL"`
+		Shards []dbSector `gorm:"constraint:OnDelete:CASCADE"` // CASCADE to delete shards too
```

Comment:
more and more I'm wondering whether we have to use this opportunity to completely detach from the 'Shard' term, I'm thinking that erasure coding is a good cutoff point, does it really make sense to keep talking about `Shards` on a database level?

I already refactored the upload code now that you got rid of `shards`... I'm just wondering whether we have to go all out now that we're getting rid of it on a database level.